src/io.rs

```diff
@@ -0,0 +1,82 @@
+// Rust Bitcoin Library
+// Written in 2014 by
+//   Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+//! no_std polyfill 
+
+#[cfg(feature = "bare-io")]
+pub use bare_io::*;
+
+#[cfg(feature = "std")]
+pub use std::io::*;
+
+#[cfg(feature = "bare-io")]
+/// A simple Write-like trait to be used for consensus encoding in a no_std environment
```

Comment:
Can't we do this without `bare-io`? It would involve swallowing the `io::Result`s and instead returning something opaque, so it kinda sucks in the sense that you end up not seeing what the error actually *was*, but it avoids needing yet another dependency just to get some simple structs. I really wish `io::ErrorKind` were public even in no-std, but maybe it makes sense to have some mapped version?

Reply:
`bare-io` is used for other things too - e.g. `Cursor`.  I can look at just grabbing a minimal implementation that does just enough for our purposes, it might be pretty small.  Is that what you mean?

Reply:
https://github.com/rust-bitcoin/bitcoin_hashes/pull/126 has an interesting approach too.

Reply:
Latest commits rip out bare-io and replace it with a small IO crate replacement.  For discussion...